proposals/browser-nodes-can-connect-to-any-node-out-of-the-box.md

```diff
@@ -0,0 +1,120 @@
+-->
+<!--
+For ease of discussion in PRs, consider breaking lines after every sentence or long phrase.
+-->
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives. 
+-->
+
+One of the key environments for the web3 is the browser. Browser nodes have several connectivity challenges that result from their environment, with the most relevant being:
+- unable to listen for incoming dials;
+- no support for TCP;
+- security policies require dials with WSS and DNS addresses.
+
+Circuit relays have been used to work around these limitations. Browser nodes can establish a connection with them and bind to them to receive incoming dials. They can also be used to "translate" a connection from Websockets to TCP and consequently, enable browser nodes to dial GO nodes.
+
+##### Problem Statement
+
+One of the critical pieces in the puzzle is the lack of ability from browser nodes to connect with GO nodes out of the box (majority of the network). By default, GO nodes do not listen on Websockets, nor have DNS + SSL configured. In addition, GO nodes do not bind to relays out of the box. This creates a network fragmentations and results in problems creating pubsub overlay, getting IPNS records, among others.
+
+Having browser nodes to connect with go nodes out of the box essentially means one of:
+- go nodes listen on webrtc by default + we can use limited relays to also negotiate webrtc conns + we deploy nodes that can act as limited relay and have dns + wss addresses (which means not all DHT servers will be capable of doing so out of the box)
+- go nodes listen on websockets address by default + we find a solution for out of the box certificates for announcing dns+wss multiaddr
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Dapp developers want browser nodes to exchange pubsub messages across the public network
+- Dapp developers want to find IPNS records on the network
+- Dapp developers want to fetch content from GO nodes
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+We had some really good discussions yesterday about the state of Pubsub/IPNS in each runtime environment  + what are the gaps for building dapps on our stack out of the box.
```

Comment:
Are there notes from the conversation?  I am seeing now that this is WIP.

Reply:
Yes, this was a wrong copy paste and not intended for the actual proposal. But here are the notes: https://hackmd.io/@gozala/state-of-ipns